### PR TITLE
Fix Switch for 128bit integers

### PR DESCRIPTION
**Pull request by bjorn3**
Fixes https://github.com/bjorn3/rustc_codegen_cranelift/issues/757
Required by https://github.com/bjorn3/rustc_codegen_cranelift/issues/743

### CODE REVIEW

File: cranelift/frontend/src/switch.rs [hunk 1]
```diff
@@ -241,6 +264,13 @@ impl Switch {
         cases_and_jt_blocks: Vec<(EntryIndex, Block, Vec<Block>)>,
     ) {
         for (first_index, jt_block, blocks) in cases_and_jt_blocks.into_iter().rev() {
+            // There are currently no 128bit systems supported by rustc, but once we do ensure that
+            // we don't silently ignore a part of the jump table for 128bit integers on 128bit systems.
+            assert!(
```
Comment (cfallin) [inline cranelift/frontend/src/switch.rs:269]: Can we just assert that the values are `<= i64::MAX` (or use `i64::try_from` below) instead? It seems fine to me to build on a 128-bit system and generate 64-bit-sized IR as long as the statically-known values are in range (unless I'm missing some other issue!).

File: cranelift/frontend/src/switch.rs [hunk 2]
```diff
@@ -333,6 +304,30 @@ impl Switch {
     }
 }
 
+fn icmp_imm_u128_eq(bx: &mut FunctionBuilder, x: Value, y: u128) -> Value {
```
Comment (cfallin) [inline cranelift/frontend/src/switch.rs:307]: Sorry to be picky here -- can we pass in an `IntCC` condition and then combine `icmp_imm_u128_{eq,uge}`?

Comment (bjorn3) [reply cranelift/frontend/src/switch.rs:307]: Already realized that :)

File: cranelift/frontend/src/switch.rs (hunk 1 above)
Comment (bjorn3) [reply cranelift/frontend/src/switch.rs:240]: Changed this to assert that `blocks.len()` fits in an `u64`.

**Comment by bjorn3** [comment]
Should I squash?

**Comment by cfallin** [comment]
Sure, go ahead and squash -- thanks!

**Comment by bjorn3** [comment]
Splat

**Pull request merged**